### ISSUE TITLE
Roaring bitmap using more storage than normal bitset

### ISSUE THREAD

**Issue by maneeshbhunwal123** [RoaringBitmap/RoaringBitmap#566]
**Describe the bug**
Roaring bitmap using more storage than normal bitset

**To Reproduce**
```
    public static void main(String[] args) throws IOException {
        RoaringBitmap roaringBitMap = new RoaringBitmap();
        BitSet bitSet = new BitSet(5000);
        double prob = 0.001;
        Random random = new Random();
        for (int i = 0; i < 5000; i++) {
            if (random.nextDouble() < prob) {
                bitSet.set(i);
                roaringBitMap.add(i);
            }
        }
        System.out.println(bitSet.cardinality());
        System.out.println("bitset bytes: "+ bitSet.size());
        System.out.println("RoaringBitmap bytes: " + roaringBitMap.getSizeInBytes() * 8);
    }
```
**RoaringBitmap version:**

0.9.0

**Java version:**
8

** description **

I have a bitset which i am using to track whether an item is present or not example

b = 01100110000

it represents that 2nd and 3rd items are present and 1st and 4th item are not present.

While searching for library which can optimise this bitset array. I came across [Roaring bitmaps][1]  which sounded very exciting. 

I did a quick test with it, 

```
    public static void main(String[] args) throws IOException {
        RoaringBitmap roaringBitMap = new RoaringBitmap();
        BitSet bitSet = new BitSet(5000);
        double prob = 0.001;
        Random random = new Random();
        for (int i = 0; i < 5000; i++) {
            if (random.nextDouble() < prob) {
                bitSet.set(i);
                roaringBitMap.add(i);
            }
        }
        System.out.println(bitSet.cardinality());
        System.out.println("bitset bytes: "+ bitSet.size());
        System.out.println("RoaringBitmap bytes: " + roaringBitMap.getSizeInBytes() * 8);
    }
```
Basically we are setting some values and check overall size of data structure.

when we run this with multiple prob values. I got 

| prob byte |  bitset bytes | RoaringBitmap bytes |
|---------------------|------------------| -- |
| 0.001 | 5056 | 288 |
| 0.01 | 5056 | 944 |
| 0.1 | 5056 | 7872 |
| 0.999 | 5056 | 65616 |

If you see as we insert more and more numbers, the memory footprint of RoaringBitmap increases. 

1. Is this expected? 
2. In the worst case should it not just fall back to bitset based implementaiton? 
3. can't 0.999 be treated as inverse of 0.001 and we would be able to store it in 288 bytes?
4. What is the most optimal way to represent these bitset as String when we are making inter service calls and using jackson library (but not byte based serialisation libraries)

**Comment by lemire** [comment]
We refer the author of the issue to the RoaringBitmap documentation starting with the [roaring specification](https://github.com/RoaringBitmap/RoaringFormatSpec). Please refer to our papers for various experimental assessments.

- Daniel Lemire, Owen Kaser, Nathan Kurz, Luca Deri, Chris O'Hara, François Saint-Jacques, Gregory Ssi-Yan-Kai, Roaring Bitmaps: Implementation of an Optimized Software Library, Software: Practice and Experience 48 (4), 2018 [arXiv:1709.07821](https://arxiv.org/abs/1709.07821)
-  Samy Chambi, Daniel Lemire, Owen Kaser, Robert Godin, Better bitmap performance with Roaring bitmaps,
Software: Practice and Experience Volume 46, Issue 5, pages 709–719, May 2016 http://arxiv.org/abs/1402.6407 This paper used data from http://lemire.me/data/realroaring2014.html
- Daniel Lemire, Gregory Ssi-Yan-Kai, Owen Kaser, Consistently faster and smaller compressed bitmaps with Roaring, Software: Practice and Experience 46 (11), 2016. http://arxiv.org/abs/1603.06549
- Samy Chambi, Daniel Lemire, Robert Godin, Kamel Boukhalfa, Charles Allen, Fangjin Yang, Optimizing Druid with Roaring bitmaps, IDEAS 2016, 2016. http://r-libre.teluq.ca/950/

**Comment by lemire** [comment]
Our README has a relevant section:

<img width="857" alt="Screen Shot 2022-06-28 at 10 56 31 AM" src="https://user-images.githubusercontent.com/391987/176211697-b1ce26be-d181-43f7-bbb4-cccd834d08c6.png">

We do not seek to make conventional bitsets obsolete. There are several instances where they are the right tool.

The submitter should assess the roaring format with their actual data. It may be that the format does not prove useful for their use case.
